packages/kbn-search-connectors/types/connectors.ts

```diff
@@ -6,6 +6,8 @@
  * Side Public License, v 1.
  */
 
+export type ConnectorType = 'elastic-crawler' | 'connector';
```

Comment:
question (same as below): do we really need this as a type? We're using connectorType and connectorMode a lot throughout this code, but it mostly makes it harder to read. Wouldn't booleans (crawler/connector and native/client) make this easier?